### PR TITLE
Add C implementation for `@stdlib/math/base/special/ellipk`

### LINKED ISSUE #766: [RFC]: Add C implementation for `@stdlib/math/base/special/ellipk`
### Description

This RFC proposes adding C implementation for [`@stdlib/math/base/special/ellipk`](https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/math/base/special/ellipk/).

```c
double stdlib_base_ellipk( const double m );
```

### Related Issues

None.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Pranavchiku**
Resolves #766.

## Checklist

- [x] update readme.md
- [x] include.gypi
- [x] binding.gyp
- [x] include/stdlib/math/base/special/
- [x] evalpoly.js
- [x] src
- [x] manifest.json
- [x] lib
- [x] examples
- [x] benchmark
- [x] test
* * *

@kgryte

### CODE REVIEW

File: lib/node_modules/@stdlib/math/base/special/ellipk/lib/native.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+var addon = require( './../src/addon.node' );
+
+
+// MAIN //
+
+/**
+* Computes the complete elliptic integral of the first kind.
+*
+* @private
+
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/ellipk/lib/native.js]:
```suggestion

```

**Comment by Pranavchiku** [comment]
All tests and benchmarks cleared, this PR can have a final review.

File: lib/node_modules/@stdlib/math/base/special/ellipk/README.md [hunk 2]
```diff
@@ -110,6 +110,96 @@ for ( i = 0; i < 100; i++ ) {
...
+#include <stdlib.h>
+#include <stdio.h>
+
+int main() {
+    double m;
+    double v;
+    int i;
+    
+    for ( i = 0; i < 100; i++ ) {
+        m = -1.0 + ( (double)rand() / (double)RAND_MAX ) * 2.0;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/ellipk/README.md]:
```suggestion
        m = -1.0 + ( ( (double)rand() / (double)RAND_MAX ) * 2.0 );
```

File: lib/node_modules/@stdlib/math/base/special/ellipk/benchmark/cpp/native/Makefile [hunk 3]
```diff
@@ -0,0 +1,146 @@
+#/
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/ellipk/benchmark/cpp/native/Makefile:1]: This file is in the wrong directory. Will fix on my end.

File: lib/node_modules/@stdlib/math/base/special/ellipk/benchmark/cpp/native/benchmark.c [hunk 4]
```diff
@@ -0,0 +1,136 @@
+/**
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/ellipk/benchmark/cpp/native/benchmark.c:1]: This file is in the wrong directory. Will fix on my end.

File: lib/node_modules/@stdlib/math/base/special/ellipk/examples/c/example.c [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+#include <stdlib.h>
+#include <stdio.h>
+
+int main() {
+    double m;
+    double v;
+    int i;
+    
+    for ( i = 0; i < 100; i++ ) {
+        m = -1.0 + ( (double)rand() / (double)RAND_MAX ) * 2.0;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/ellipk/examples/c/example.c]:
```suggestion
        m = -1.0 + ( ( (double)rand() / (double)RAND_MAX ) * 2.0 );
```

File: lib/node_modules/@stdlib/math/base/special/ellipk/src/main.c [hunk 6]
```diff
@@ -0,0 +1,366 @@
...
+*
+* @param x    input value
+* @return     output value
+*
+* @example
+* double out = stdlib_base_ellipk( 0.5 );
+* // returns ~1.854
+*/
+double stdlib_base_ellipk( const double m ) {
+    int32_t FLG = 0;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/ellipk/src/main.c]:
```suggestion
    int8_t FLG = 0;
```

File: lib/node_modules/@stdlib/math/base/special/ellipk/src/main.c [hunk 7]
```diff
@@ -0,0 +1,366 @@
...
+        t = poly_p9( x - 0.825 );
+    } else if ( x < 0.9 ) {
+        t = poly_p10( x - 0.875 );
+    } else {
+        td = 1.0 - x;
+        qd = poly_p11( td );
+        kdm = poly_p12( td - 0.05 );
+        t = -stdlib_base_ln( qd ) * ( kdm * ONE_DIV_PI ); 
+    }
+    if ( FLG == 1) {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/ellipk/src/main.c]:
```suggestion
    if ( FLG == 1 ) {
```

**Review by kgryte — APPROVED** [review]
LGTM. Thanks, @Pranavchiku!

**Pull request merged**